### PR TITLE
[tests] Update tests to increase coverage

### PR DESCRIPTION
**Pull request by vchrombie**
This PR adds some more tests that target some missing lines to increase the coverage.

### CODE REVIEW

File: tests/test_publish.py [hunk 1]
```diff
@@ -195,6 +198,41 @@ def test_publish_to_remote(self, mock_project, mock_read_changelog):
...
+            result = runner.invoke(publish.publish,
+                                   ["0.8.10", "John Smith <jsmith@example.org>"])
+            self.assertEqual(result.exit_code, 1)
+
+            lines = result.stderr.split('\n')
+            self.assertRegex(lines[-2], MOCK_REPOSITORY_ERROR)
+
+    @unittest.mock.patch('release_tools.publish.remove_unreleased_changelog_entries')
+    @unittest.mock.patch('release_tools.publish.Project')
+    def test_publish_repository_error_2(self, mock_project, mock_remove_unreleased_changelog):
```
Comment (sduenas) [inline tests/test_publish.py:220]: The title of this test should explain better where the problem and what it's really testing.

File: tests/test_version.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
```
Comment (sduenas) [inline tests/test_version.py:1]: This shouldn't be tested.

**Review by sduenas — CHANGES_REQUESTED** [review]
In general it looks good. Check my comments, though.

File: tests/test_version.py (hunk 2 above)
Comment (vchrombie) [reply tests/test_version.py:1]: Okay :+1:

File: tests/test_publish.py [hunk 3]
```diff
@@ -195,6 +198,41 @@ def test_publish_to_remote(self, mock_project, mock_read_changelog):
...
+            result = runner.invoke(publish.publish,
+                                   ["0.8.10", "John Smith <jsmith@example.org>"])
+            self.assertEqual(result.exit_code, 1)
+
+            lines = result.stderr.split('\n')
+            self.assertRegex(lines[-2], MOCK_REPOSITORY_ERROR)
+
+    @unittest.mock.patch('release_tools.publish.remove_unreleased_changelog_entries')
+    @unittest.mock.patch('release_tools.publish.Project')
+    def test_remove_unreleased_changelog_entries_repository_error(self, mock_project, mock_remove_unreleased_changelog):
```
Comment (sduenas) [inline tests/test_publish.py:220]: Would you mind to split this line? I think it's very long.

**Review by sduenas — APPROVED** [review]
LGTM

**Pull request merged**